packages/headless/src/features/attached-results/attached-results-analytics-actions.ts

```diff
@@ -55,3 +60,51 @@ export const logCaseDetach = (result: Result) =>
       };
     },
   });
+
+export const logCitationDocumentAttach = (citation: GeneratedAnswerCitation) =>
+  makeInsightAnalyticsActionFactory(
+    SearchPageEvents.generatedAnswerCitationDocumentAttach
+  )({
+    prefix: 'insight/generatedAnswerCitationDocumentAttach',
+    __legacy__getBuilder: (client, state) => {
+      const metadata = getCaseContextAnalyticsMetadata(
+        state.insightCaseContext
+      );
+      const generativeQuestionAnsweringId =
+        generativeQuestionAnsweringIdSelector(state);
+
+      if (!generativeQuestionAnsweringId || !citation) {
+        return null;
+      }
+
+      const citationPayload = {
+        generativeQuestionAnsweringId,
+        citationId: citation.id,
+        documentId: citationDocumentIdentifier(citation),
+      };
+      return client.logGeneratedAnswerCitationDocumentAttach(
+        partialCitationInformation(citation, state),
+        citationPayload,
+        metadata
+      );
+    },
+  });
+
+export const logCitationDocumentDetach = (citation: GeneratedAnswerCitation) =>
+  makeInsightAnalyticsActionFactory(SearchPageEvents.caseDetach)({
+    prefix: 'insight/caseDetach',
+    __legacy__getBuilder: (client, state) => {
+      const uriHash = citation.fields?.urihash || '';
+      return client.logCaseDetach(
+        uriHash,
+        getCaseContextAnalyticsMetadata(state.insightCaseContext)
+      );
+    },
```

Comment:
```suggestion
    __legacy__getBuilder: (client, state) => {
      const uriHash = citation.fields?.urihash || '';
      return client.logCaseDetach(
        uriHash,
        citation.permanentId,
        getCaseContextAnalyticsMetadata(state.insightCaseContext)
      );
    },
```

we need to add this here once the coveo UA is updated